### ISSUE TITLE
Programmatic servlet mappings cannot override mappings from webdefault.xml using quickstart

### ISSUE THREAD

**Issue by Antibrumm** [eclipse/jetty.project#877]
Hi
I don't know which information I would need to provide to identify the problem so just let me know.

I have a multi-module maven project where one of the modules is an intergration-test module simulating the web application. This web application is started upfront the protractor tests and then shutdown again with the help of the jetty-maven-plugin. No this is all running well when using the `start` and `stop` goals. 

Finally I wanted to try using `run-forked` to activate the jacoco javaagent, but this does not seem to produce a running application. 

As a test I tried using the `run` goal and to my surprise this produced a working web application.

I activated some debug logging for both and could verify that the application is started in both cases: My logs from the application startup show that the system is actually booting fine, but in the end, for the `run-forked` goal there is only a "browse directory" screen, while with the `run` goal the context is setup correctly and I can access the pages.

What i found out until now is that the `JettyWebAppContext` look different.
- While with the `run`goal the context uses absolute file uris, this is not the case with the `run-forked` goal.
- Also the context looks a lot bigger in the second case, which i cannot explain.

```
Started o.e.j.m.p.JettyWebAppContext@a357c3e {/webapp,[file:///jd/ws/texelia/webapp/webapp/base/ui-angular/tests/src/test/webapp/, jar:file:///Users/mfrey/.m2/repository/org/webjars/jQuery-slimScroll/1.3.3/jQuery-slimScroll-1.3.3.jar!/META-INF/resources, jar:file:///Users/mfrey/.m2/repository/org/webjars/angularjs/1.5.5/angularjs-1.5.5.jar!/META-INF/resources, jar:file:///Users/mfrey/.m2/repository/org/webjars/angular-ui-bootstrap/1.3.2/angular-ui-bootstrap-1.3.2.jar!/META-INF/resources, jar:file:///Users/mfrey/.m2/repository/org/webjars/toastr/2.1.0/toastr-2.1.0.jar!/META-INF/resources, jar:file:///Users/mfrey/.m2/repository/org/webjars/jquery/2.2.4/jquery-2.2.4.jar!/META-INF/resources, jar:file:///Users/mfrey/.m2/repository/org/webjars/bootstrap/3.3.6/bootstrap-3.3.6.jar!/META-INF/resources, jar:file:///Users/mfrey/.m2/repository/org/webjars/jquery-easing/1.3-1/jquery-easing-1.3-1.jar!/META-INF/resources, jar:file:///Users/mfrey/.m2/repository/org/webjars/dropzone/3.12.0/dropzone-3.12.0.jar!/META-INF/resources, jar:file:///Users/mfrey/.m2/repository/org/webjars/jquery.fileDownload/50171edfab/jquery.fileDownload-50171edfab.jar!/META-INF/resources, jar:file:///Users/mfrey/.m2/repository/org/webjars/jasmine/2.0.0/jasmine-2.0.0.jar!/META-INF/resources],AVAILABLE}{file:///jd/ws/texelia/webapp/webapp/base/ui-angular/tests/src/test/webapp/}
```

```
Started o.e.j.m.p.JettyWebAppContext@400d912a{/webapp,[file:///jd/ws/texelia/webapp/webapp/base/ui-angular/tests/src/test/webapp/, jar:file:/Users/mfrey/.m2/repository/org/webjars/jQuery-slimScroll/1.3.3/jQuery-slimScroll-1.3.3.jar!/META-INF/resources, jar:file:/Users/mfrey/.m2/repository/org/webjars/angularjs/1.5.5/angularjs-1.5.5.jar!/META-INF/resources,   jar:file:/Users/mfrey/.m2/repository/org/webjars/jQuery-slimScroll/1.3.3/jQuery-slimScroll-1.3.3.jar!/META-INF/resources,  jar:file:/Users/mfrey/.m2/repository/org/webjars/angular-ui-bootstrap/1.3.2/angular-ui-bootstrap-1.3.2.jar!/META-INF/resources, jar:file:/Users/mfrey/.m2/repository/org/webjars/angularjs/1.5.5/angularjs-1.5.5.jar!/META-INF/resources, jar:file:/Users/mfrey/.m2/repository/org/webjars/jQuery-slimScroll/1.3.3/jQuery-slimScroll-1.3.3.jar!/META-INF/resources, jar:file:/Users/mfrey/.m2/repository/org/webjars/angular-ui-bootstrap/1.3.2/angular-ui-bootstrap-1.3.2.jar!/META-INF/resources, jar:file:/Users/mfrey/.m2/repository/org/webjars/toastr/2.1.0/toastr-2.1.0.jar!/META-INF/resources, jar:file:/Users/mfrey/.m2/repository/org/webjars/angularjs/1.5.5/angularjs-1.5.5.jar!/META-INF/resources, jar:file:/Users/mfrey/.m2/repository/org/webjars/jQuery-slimScroll/1.3.3/jQuery-slimScroll-1.3.3.jar!/META-INF/resources, jar:file:/Users/mfrey/.m2/repository/org/webjars/angular-ui-bootstrap/1.3.2/angular-ui-bootstrap-1.3.2.jar!/META-INF/resources, jar:file:/Users/mfrey/.m2/repository/org/webjars/toastr/2.1.0/toastr-2.1.0.jar!/META-INF/resources, jar:file:/Users/mfrey/.m2/repository/org/webjars/angularjs/1.5.5/angularjs-1.5.5.jar!/META-INF/resources, jar:file:/Users/mfrey/.m2/repository/org/webjars/jquery/2.2.4/jquery-2.2.4.jar!/META-INF/resources, jar:file:/Users/mfrey/.m2/repository/org/webjars/jQuery-slimScroll/1.3.3/jQuery-slimScroll-1.3.3.jar!/META-INF/resources, jar:file:/Users/mfrey/.m2/repository/org/webjars/angular-ui-bootstrap/1.3.2/angular-ui-bootstrap-1.3.2.jar!/META-INF/resources, jar:file:/Users/mfrey/.m2/repository/org/webjars/toastr/2.1.0/toastr-2.1.0.jar!/META-INF/resources, jar:file:/Users/mfrey/.m2/repository/org/webjars/angularjs/1.5.5/angularjs-1.5.5.jar!/META-INF/resources, jar:file:/Users/mfrey/.m2/repository/org/webjars/jquery/2.2.4/jquery-2.2.4.jar!/META-INF/resources, jar:file:/Users/mfrey/.m2/repository/org/webjars/bootstrap/3.3.6/bootstrap-3.3.6.jar!/META-INF/resources, jar:file:/Users/mfrey/.m2/repository/org/webjars/jQuery-slimScroll/1.3.3/jQuery-slimScroll-1.3.3.jar!/META-INF/resources, jar:file:/Users/mfrey/.m2/repository/org/webjars/angular-ui-bootstrap/1.3.2/angular-ui-bootstrap-1.3.2.jar!/META-INF/resources, jar:file:/Users/mfrey/.m2/repository/org/webjars/toastr/2.1.0/toastr-2.1.0.jar!/META-INF/resources, jar:file:/Users/mfrey/.m2/repository/org/webjars/angularjs/1.5.5/angularjs-1.5.5.jar!/META-INF/resources, jar:file:/Users/mfrey/.m2/repository/org/webjars/jquery-easing/1.3-1/jquery-easing-1.3-1.jar!/META-INF/resources, jar:file:/Users/mfrey/.m2/repository/org/webjars/jquery/2.2.4/jquery-2.2.4.jar!/META-INF/resources, jar:file:/Users/mfrey/.m2/repository/org/webjars/bootstrap/3.3.6/bootstrap-3.3.6.jar!/META-INF/resources, jar:file:/Users/mfrey/.m2/repository/org/webjars/jQuery-slimScroll/1.3.3/jQuery-slimScroll-1.3.3.jar!/META-INF/resources, jar:file:/Users/mfrey/.m2/repository/org/webjars/angular-ui-bootstrap/1.3.2/angular-ui-bootstrap-1.3.2.jar!/META-INF/resources, jar:file:/Users/mfrey/.m2/repository/org/webjars/toastr/2.1.0/toastr-2.1.0.jar!/META-INF/resources, jar:file:/Users/mfrey/.m2/repository/org/webjars/angularjs/1.5.5/angularjs-1.5.5.jar!/META-INF/resources, jar:file:/Users/mfrey/.m2/repository/org/webjars/jquery-easing/1.3-1/jquery-easing-1.3-1.jar!/META-INF/resources, jar:file:/Users/mfrey/.m2/repository/org/webjars/jquery/2.2.4/jquery-2.2.4.jar!/META-INF/resources, jar:file:/Users/mfrey/.m2/repository/org/webjars/bootstrap/3.3.6/bootstrap-3.3.6.jar!/META-INF/resources, jar:file:/Users/mfrey/.m2/repository/org/webjars/dropzone/3.12.0/dropzone-3.12.0.jar!/META-INF/resources, jar:file:/Users/mfrey/.m2/repository/org/webjars/jQuery-slimScroll/1.3.3/jQuery-slimScroll-1.3.3.jar!/META-INF/resources, jar:file:/Users/mfrey/.m2/repository/org/webjars/angular-ui-bootstrap/1.3.2/angular-ui-bootstrap-1.3.2.jar!/META-INF/resources, jar:file:/Users/mfrey/.m2/repository/org/webjars/toastr/2.1.0/toastr-2.1.0.jar!/META-INF/resources, jar:file:/Users/mfrey/.m2/repository/org/webjars/angularjs/1.5.5/angularjs-1.5.5.jar!/META-INF/resources, jar:file:/Users/mfrey/.m2/repository/org/webjars/jquery-easing/1.3-1/jquery-easing-1.3-1.jar!/META-INF/resources, jar:file:/Users/mfrey/.m2/repository/org/webjars/jquery/2.2.4/jquery-2.2.4.jar!/META-INF/resources, jar:file:/Users/mfrey/.m2/repository/org/webjars/bootstrap/3.3.6/bootstrap-3.3.6.jar!/META-INF/resources, jar:file:/Users/mfrey/.m2/repository/org/webjars/jquery.fileDownload/50171edfab/jquery.fileDownload-50171edfab.jar!/META-INF/resources, jar:file:/Users/mfrey/.m2/repository/org/webjars/dropzone/3.12.0/dropzone-3.12.0.jar!/META-INF/resources, jar:file:/Users/mfrey/.m2/repository/org/webjars/jQuery-slimScroll/1.3.3/jQuery-slimScroll-1.3.3.jar!/META-INF/resources, jar:file:/Users/mfrey/.m2/repository/org/webjars/angular-ui-bootstrap/1.3.2/angular-ui-bootstrap-1.3.2.jar!/META-INF/resources, jar:file:/Users/mfrey/.m2/repository/org/webjars/toastr/2.1.0/toastr-2.1.0.jar!/META-INF/resources, jar:file:/Users/mfrey/.m2/repository/org/webjars/jasmine/2.0.0/jasmine-2.0.0.jar!/META-INF/resources, jar:file:/Users/mfrey/.m2/repository/org/webjars/angularjs/1.5.5/angularjs-1.5.5.jar!/META-INF/resources, jar:file:/Users/mfrey/.m2/repository/org/webjars/jquery-easing/1.3-1/jquery-easing-1.3-1.jar!/META-INF/resources, jar:file:/Users/mfrey/.m2/repository/org/webjars/jquery/2.2.4/jquery-2.2.4.jar!/META-INF/resources, jar:file:/Users/mfrey/.m2/repository/org/webjars/bootstrap/3.3.6/bootstrap-3.3.6.jar!/META-INF/resources, jar:file:/Users/mfrey/.m2/repository/org/webjars/jquery.fileDownload/50171edfab/jquery.fileDownload-50171edfab.jar!/META-INF/resources, jar:file:/Users/mfrey/.m2/repository/org/webjars/dropzone/3.12.0/dropzone-3.12.0.jar!/META-INF/resources, jar:file:/Users/mfrey/.m2/repository/org/webjars/jQuery-slimScroll/1.3.3/jQuery-slimScroll-1.3.3.jar!/META-INF/resources],AVAILABLE}
```

**Comment by joakime** [comment]
Others have reported similar issues on stackoverflow.com

 - http://stackoverflow.com/questions/39228668/jacoco-and-jetty-maven-plugin-gets-0-coverage-in-sonarqube
 - http://stackoverflow.com/questions/33645713/jacoco-and-jetty-maven-plugin-gets-0-coverage

This accepted answer has helped others doing the same thing you are attempting.

- http://stackoverflow.com/a/35413159/775715

**Comment by joakime** [comment]
@daniellundmark has an example project that shows this setup in operation.

https://github.com/daniellundmark/jetty-jacoco-example

**Comment by Antibrumm** [comment]
Thanks for the quick feedback.

Yes i saw these answers too and was working on this approach already :) I havent succeeded yet however, as the test module is not really a war probably. Will work in this direction again. 

I still wanted to report it as I think this is not the expected behavior for the run-forked goal. I would be happy to provide you with more information, like debug logs or whatever, if you feel like it is worth investigating. My project is not public but I should be able to strip out the important parts.

**Comment by janbartel** [comment]
@Antibrumm I would need a small repro test case to be able to really diagnose any problem with the jetty plugin.

However, I do think the duplicate jars on the webapp classpath are odd. I can only assume that as it is a multi-module build, maven lists these jars as project artifacts multiple times. Can you run with -X flag and report how many lines of the form  "Adding artifact XXXX with scope YYY for WEB-INF/lib " you see?

**Comment by Antibrumm** [comment]
```
#properties for forked webapp
#Wed Aug 31 07:42:57 CEST 2016
tmp.dir.persist=false
base.dirs.orig=file\:///jd/ws/company/webapp/webapp/base/ui-angular/tests/src/main/webapp/
classes.dir=/jd/ws/company/webapp/webapp/base/ui-angular/tests/target/classes
lib.jars=/Users/mfrey/.m2/repository/com/company/webapp/ui/angular/core-angular/1.5.0.0-SNAPSHOT/core-angular-1.5.0.0-SNAPSHOT.jar,/Users/mfrey/.m2/repository/com/company/webapp/ui/rest/core-rest/1.5.0.0-SNAPSHOT/core-rest-1.5.0.0-SNAPSHOT.jar,/Users/mfrey/.m2/repository/com/company/webapp/core/core/1.5.0.0-SNAPSHOT/core-1.5.0.0-SNAPSHOT.jar,/Users/mfrey/.m2/repository/com/company/webapp/core/model/1.5.0.0-SNAPSHOT/model-1.5.0.0-SNAPSHOT.jar,/Users/mfrey/.m2/repository/com/company/webapp/core/modules-api/1.5.0.0-SNAPSHOT/modules-api-1.5.0.0-SNAPSHOT.jar,/Users/mfrey/.m2/repository/org/atteo/evo-inflector/1.2.1/evo-inflector-1.2.1.jar,/Users/mfrey/.m2/repository/org/jasypt/jasypt/1.9.2/jasypt-1.9.2.jar,/Users/mfrey/.m2/repository/ch/mfrey/jackson/jackson-antpathfilter/1.0.2/jackson-antpathfilter-1.0.2.jar,/Users/mfrey/.m2/repository/org/webjars/jquery/2.2.4/jquery-2.2.4.jar,/Users/mfrey/.m2/repository/org/webjars/angularjs/1.5.5/angularjs-1.5.5.jar,/Users/mfrey/.m2/repository/org/webjars/bootstrap/3.3.6/bootstrap-3.3.6.jar,/Users/mfrey/.m2/repository/org/webjars/angular-ui-bootstrap/1.3.2/angular-ui-bootstrap-1.3.2.jar,/Users/mfrey/.m2/repository/org/webjars/toastr/2.1.0/toastr-2.1.0.jar,/Users/mfrey/.m2/repository/org/webjars/jQuery-slimScroll/1.3.3/jQuery-slimScroll-1.3.3.jar,/Users/mfrey/.m2/repository/org/webjars/jquery-easing/1.3-1/jquery-easing-1.3-1.jar,/Users/mfrey/.m2/repository/org/webjars/dropzone/3.12.0/dropzone-3.12.0.jar,/Users/mfrey/.m2/repository/org/webjars/jquery.fileDownload/50171edfab/jquery.fileDownload-50171edfab.jar,/Users/mfrey/.m2/repository/com/github/spullara/mustache/java/compiler/0.9.0/compiler-0.9.0.jar,/Users/mfrey/.m2/repository/org/webjars/jasmine/2.0.0/jasmine-2.0.0.jar,/Users/mfrey/.m2/repository/org/hsqldb/hsqldb/2.3.3/hsqldb-2.3.3.jar,/Users/mfrey/.m2/repository/com/company/webapp/ui/angular/auth-db-angular/1.5.0.0-SNAPSHOT/auth-db-angular-1.5.0.0-SNAPSHOT.jar,/Users/mfrey/.m2/repository/com/company/webapp/ui/rest/auth-db-rest/1.5.0.0-SNAPSHOT/auth-db-rest-1.5.0.0-SNAPSHOT.jar,/Users/mfrey/.m2/repository/com/company/webapp/core/modules/auth-db/1.5.0.0-SNAPSHOT/auth-db-1.5.0.0-SNAPSHOT.jar,/Users/mfrey/.m2/repository/com/company/webapp/core/modules/mail-api/1.5.0.0-SNAPSHOT/mail-api-1.5.0.0-SNAPSHOT.jar,/Users/mfrey/.m2/repository/com/company/webapp/ui/angular/auth-ldap-angular/1.5.0.0-SNAPSHOT/auth-ldap-angular-1.5.0.0-SNAPSHOT.jar,/Users/mfrey/.m2/repository/com/company/webapp/ui/rest/auth-ldap-rest/1.5.0.0-SNAPSHOT/auth-ldap-rest-1.5.0.0-SNAPSHOT.jar,/Users/mfrey/.m2/repository/com/company/webapp/core/modules/auth-ldap/1.5.0.0-SNAPSHOT/auth-ldap-1.5.0.0-SNAPSHOT.jar,/Users/mfrey/.m2/repository/org/springframework/security/spring-security-ldap/4.1.0.RELEASE/spring-security-ldap-4.1.0.RELEASE.jar,/Users/mfrey/.m2/repository/org/springframework/ldap/spring-ldap-core/2.0.2.RELEASE/spring-ldap-core-2.0.2.RELEASE.jar,/Users/mfrey/.m2/repository/com/company/webapp/ui/angular/attachments-file-angular/1.5.0.0-SNAPSHOT/attachments-file-angular-1.5.0.0-SNAPSHOT.jar,/Users/mfrey/.m2/repository/com/company/webapp/ui/rest/attachments-file-rest/1.5.0.0-SNAPSHOT/attachments-file-rest-1.5.0.0-SNAPSHOT.jar,/Users/mfrey/.m2/repository/com/company/webapp/core/modules/attachments-file/1.5.0.0-SNAPSHOT/attachments-file-1.5.0.0-SNAPSHOT.jar,/Users/mfrey/.m2/repository/com/company/webapp/ui/angular/import-angular/1.5.0.0-SNAPSHOT/import-angular-1.5.0.0-SNAPSHOT.jar,/Users/mfrey/.m2/repository/com/company/webapp/ui/rest/import-rest/1.5.0.0-SNAPSHOT/import-rest-1.5.0.0-SNAPSHOT.jar,/Users/mfrey/.m2/repository/com/company/webapp/core/modules/import/1.5.0.0-SNAPSHOT/import-1.5.0.0-SNAPSHOT.jar,/Users/mfrey/.m2/repository/com/company/webapp/core/modules/import-api/1.5.0.0-SNAPSHOT/import-api-1.5.0.0-SNAPSHOT.jar,/Users/mfrey/.m2/repository/org/apache/poi/poi/3.13/poi-3.13.jar,/Users/mfrey/.m2/repository/commons-codec/commons-codec/1.9/commons-codec-1.9.jar,/Users/mfrey/.m2/repository/org/apache/poi/poi-ooxml/3.13/poi-ooxml-3.13.jar,/Users/mfrey/.m2/repository/org/apache/poi/poi-ooxml-schemas/3.13/poi-ooxml-schemas-3.13.jar,/Users/mfrey/.m2/repository/org/apache/xmlbeans/xmlbeans/2.6.0/xmlbeans-2.6.0.jar,/Users/mfrey/.m2/repository/stax/stax-api/1.0.1/stax-api-1.0.1.jar,/Users/mfrey/.m2/repository/com/company/webapp/ui/angular/excelexport-angular/1.5.0.0-SNAPSHOT/excelexport-angular-1.5.0.0-SNAPSHOT.jar,/Users/mfrey/.m2/repository/com/company/webapp/ui/rest/excelexport-rest/1.5.0.0-SNAPSHOT/excelexport-rest-1.5.0.0-SNAPSHOT.jar,/Users/mfrey/.m2/repository/com/company/webapp/core/modules/excelexport/1.5.0.0-SNAPSHOT/excelexport-1.5.0.0-SNAPSHOT.jar,/Users/mfrey/.m2/repository/com/company/webapp/core/modules/excelexport-api/1.5.0.0-SNAPSHOT/excelexport-api-1.5.0.0-SNAPSHOT.jar,/Users/mfrey/.m2/repository/com/company/webapp/ui/angular/test-support-angular/1.5.0.0-SNAPSHOT/test-support-angular-1.5.0.0-SNAPSHOT.jar,/Users/mfrey/.m2/repository/com/company/webapp/core/tests/test-entities/1.5.0.0-SNAPSHOT/test-entities-1.5.0.0-SNAPSHOT.jar,/Users/mfrey/.m2/repository/com/company/webapp/core/modules/1.5.0.0-SNAPSHOT/modules-1.5.0.0-SNAPSHOT.jar,/Users/mfrey/.m2/repository/org/springframework/spring-context/4.3.1.RELEASE/spring-context-4.3.1.RELEASE.jar,/Users/mfrey/.m2/repository/org/springframework/spring-aop/4.3.1.RELEASE/spring-aop-4.3.1.RELEASE.jar,/Users/mfrey/.m2/repository/org/springframework/spring-beans/4.3.1.RELEASE/spring-beans-4.3.1.RELEASE.jar,/Users/mfrey/.m2/repository/org/springframework/spring-core/4.3.1.RELEASE/spring-core-4.3.1.RELEASE.jar,/Users/mfrey/.m2/repository/org/springframework/spring-expression/4.3.1.RELEASE/spring-expression-4.3.1.RELEASE.jar,/Users/mfrey/.m2/repository/org/springframework/spring-context-support/4.3.1.RELEASE/spring-context-support-4.3.1.RELEASE.jar,/Users/mfrey/.m2/repository/org/springframework/spring-web/4.3.1.RELEASE/spring-web-4.3.1.RELEASE.jar,/Users/mfrey/.m2/repository/org/springframework/spring-webmvc/4.3.1.RELEASE/spring-webmvc-4.3.1.RELEASE.jar,/Users/mfrey/.m2/repository/org/springframework/spring-orm/4.3.1.RELEASE/spring-orm-4.3.1.RELEASE.jar,/Users/mfrey/.m2/repository/org/springframework/spring-jdbc/4.3.1.RELEASE/spring-jdbc-4.3.1.RELEASE.jar,/Users/mfrey/.m2/repository/org/springframework/spring-tx/4.3.1.RELEASE/spring-tx-4.3.1.RELEASE.jar,/Users/mfrey/.m2/repository/org/springframework/security/spring-security-config/4.1.0.RELEASE/spring-security-config-4.1.0.RELEASE.jar,/Users/mfrey/.m2/repository/aopalliance/aopalliance/1.0/aopalliance-1.0.jar,/Users/mfrey/.m2/repository/org/springframework/security/spring-security-core/4.1.0.RELEASE/spring-security-core-4.1.0.RELEASE.jar,/Users/mfrey/.m2/repository/org/springframework/security/spring-security-web/4.1.0.RELEASE/spring-security-web-4.1.0.RELEASE.jar,/Users/mfrey/.m2/repository/org/aspectj/aspectjweaver/1.8.7/aspectjweaver-1.8.7.jar,/Users/mfrey/.m2/repository/org/slf4j/slf4j-api/1.7.12/slf4j-api-1.7.12.jar,/Users/mfrey/.m2/repository/org/slf4j/jcl-over-slf4j/1.7.12/jcl-over-slf4j-1.7.12.jar,/Users/mfrey/.m2/repository/ch/qos/logback/logback-classic/1.1.3/logback-classic-1.1.3.jar,/Users/mfrey/.m2/repository/ch/qos/logback/logback-core/1.1.3/logback-core-1.1.3.jar,/Users/mfrey/.m2/repository/com/fasterxml/jackson/core/jackson-core/2.6.3/jackson-core-2.6.3.jar,/Users/mfrey/.m2/repository/com/fasterxml/jackson/core/jackson-databind/2.6.3/jackson-databind-2.6.3.jar,/Users/mfrey/.m2/repository/com/fasterxml/jackson/core/jackson-annotations/2.6.0/jackson-annotations-2.6.0.jar,/Users/mfrey/.m2/repository/org/hibernate/hibernate-entitymanager/5.0.7.Final/hibernate-entitymanager-5.0.7.Final.jar,/Users/mfrey/.m2/repository/org/jboss/logging/jboss-logging/3.3.0.Final/jboss-logging-3.3.0.Final.jar,/Users/mfrey/.m2/repository/org/hibernate/hibernate-core/5.0.7.Final/hibernate-core-5.0.7.Final.jar,/Users/mfrey/.m2/repository/antlr/antlr/2.7.7/antlr-2.7.7.jar,/Users/mfrey/.m2/repository/org/jboss/jandex/2.0.0.Final/jandex-2.0.0.Final.jar,/Users/mfrey/.m2/repository/dom4j/dom4j/1.6.1/dom4j-1.6.1.jar,/Users/mfrey/.m2/repository/xml-apis/xml-apis/1.0.b2/xml-apis-1.0.b2.jar,/Users/mfrey/.m2/repository/org/hibernate/common/hibernate-commons-annotations/5.0.1.Final/hibernate-commons-annotations-5.0.1.Final.jar,/Users/mfrey/.m2/repository/org/hibernate/javax/persistence/hibernate-jpa-2.1-api/1.0.0.Final/hibernate-jpa-2.1-api-1.0.0.Final.jar,/Users/mfrey/.m2/repository/org/javassist/javassist/3.18.1-GA/javassist-3.18.1-GA.jar,/Users/mfrey/.m2/repository/org/apache/geronimo/specs/geronimo-jta_1.1_spec/1.1.1/geronimo-jta_1.1_spec-1.1.1.jar,/Users/mfrey/.m2/repository/org/hibernate/hibernate-envers/5.0.7.Final/hibernate-envers-5.0.7.Final.jar,/Users/mfrey/.m2/repository/org/hibernate/hibernate-validator/5.2.2.Final/hibernate-validator-5.2.2.Final.jar,/Users/mfrey/.m2/repository/javax/validation/validation-api/1.1.0.Final/validation-api-1.1.0.Final.jar,/Users/mfrey/.m2/repository/com/fasterxml/classmate/1.1.0/classmate-1.1.0.jar,/Users/mfrey/.m2/repository/javax/el/javax.el-api/2.2.4/javax.el-api-2.2.4.jar,/Users/mfrey/.m2/repository/org/glassfish/web/javax.el/2.2.4/javax.el-2.2.4.jar,/Users/mfrey/.m2/repository/org/hibernate/hibernate-ehcache/5.0.7.Final/hibernate-ehcache-5.0.7.Final.jar,/Users/mfrey/.m2/repository/net/sf/ehcache/ehcache-core/2.4.3/ehcache-core-2.4.3.jar,/Users/mfrey/.m2/repository/joda-time/joda-time/2.8.2/joda-time-2.8.2.jar,/Users/mfrey/.m2/repository/org/jadira/usertype/usertype.core/5.0.0.GA/usertype.core-5.0.0.GA.jar,/Users/mfrey/.m2/repository/org/jadira/usertype/usertype.spi/5.0.0.GA/usertype.spi-5.0.0.GA.jar,/Users/mfrey/.m2/repository/com/fasterxml/jackson/datatype/jackson-datatype-joda/2.6.3/jackson-datatype-joda-2.6.3.jar,/Users/mfrey/.m2/repository/commons-fileupload/commons-fileupload/1.3.1/commons-fileupload-1.3.1.jar,/Users/mfrey/.m2/repository/commons-io/commons-io/2.4/commons-io-2.4.jar,/Users/mfrey/.m2/repository/javax/annotation/javax.annotation-api/1.2/javax.annotation-api-1.2.jar,/Users/mfrey/.m2/repository/org/apache/commons/commons-lang3/3.4/commons-lang3-3.4.jar,/Users/mfrey/.m2/repository/org/springframework/spring-test/4.3.1.RELEASE/spring-test-4.3.1.RELEASE.jar,/Users/mfrey/.m2/repository/junit/junit/4.12/junit-4.12.jar,/Users/mfrey/.m2/repository/org/hamcrest/hamcrest-core/1.3/hamcrest-core-1.3.jar,/Users/mfrey/.m2/repository/xmlunit/xmlunit/1.6/xmlunit-1.6.jar,/Users/mfrey/.m2/repository/org/hamcrest/hamcrest-library/1.3/hamcrest-library-1.3.jar
context.path=/webapp
base.dirs=file\:///jd/ws/company/webapp/webapp/base/ui-angular/tests/src/main/webapp/
testClasses.dir=/jd/ws/company/webapp/webapp/base/ui-angular/tests/target/test-classes
maven.war.excludes=
maven.war.includes=
quickstart.web.xml=/jd/ws/company/webapp/webapp/base/ui-angular/tests/target/fork-web.xml
tmp.dir=/jd/ws/company/webapp/webapp/base/ui-angular/tests/target/tmp
```

**Comment by Antibrumm** [comment]
I also need to mention that the project is Spring based and i can see in the fork-web.xml the springservletinitializer:
```
  <context-param>
    <param-name>org.eclipse.jetty.containerInitializers</param-name>
    <param-value><![CDATA[
    "ContainerInitializer{org.eclipse.jetty.websocket.jsr356.server.deploy.WebSocketServerContainerInitializer,interested=[javax.websocket.server.ServerApplicationConfig, javax.websocket.server.ServerEndpoint, javax.websocket.Endpoint],applicable=[],annotated=[]}",
    "ContainerInitializer{org.eclipse.jetty.apache.jsp.JettyJasperInitializer,interested=[],applicable=[],annotated=[]}",
    "ContainerInitializer{org.springframework.web.SpringServletContainerInitializer,interested=[org.springframework.web.WebApplicationInitializer],applicable=[org.springframework.web.context.AbstractContextLoaderInitializer, com.company.webapp.core.init.root.WebApplicationRootInitializer, org.springframework.security.web.context.AbstractSecurityWebApplicationInitializer, org.springframework.web.servlet.support.AbstractAnnotationConfigDispatcherServletInitializer, com.company.webapp.rest.core.init.web.WebApplicationServletInitializer, org.springframework.web.servlet.support.AbstractDispatcherServletInitializer],annotated=[]}"]]></param-value>
  </context-param>
```

And here is the resulting page:
```
<HTML><HEAD><LINK HREF="jetty-dir.css" REL="stylesheet" TYPE="text/css"/><TITLE>Directory: /webapp/</TITLE></HEAD><BODY>
<H1>Directory: /webapp/</H1>
<TABLE BORDER=0>

<TR><TD><A HREF="/webapp/WEB-INF/">WEB-INF/&nbsp;</A></TD><TD ALIGN=right>102 bytes&nbsp;</TD><TD>Aug 29, 2016 1:42:09 PM</TD></TR>
<TR><TD><A HREF="/webapp/webjars/">webjars/&nbsp;</A></TD><TD ALIGN=right>-1 bytes&nbsp;</TD><TD>Oct 28, 2014 5:28:12 PM</TD></TR>
<TR><TD><A HREF="/webapp/webjars/angular-ui-bootstrap/">webjars/angular-ui-bootstrap/&nbsp;</A></TD><TD ALIGN=right>-1 bytes&nbsp;</TD><TD>Apr 20, 2016 7:33:46 AM</TD></TR>
....
<TR><TD><A HREF="/webapp/webjars/toastr/2.1.0/webjars-requirejs.js">webjars/toastr/2.1.0/webjars-requirejs.js&nbsp;</A></TD><TD ALIGN=right>106 bytes&nbsp;</TD><TD>Nov 11, 2014 8:04:44 AM</TD></TR></TABLE>
</BODY></HTML>
```

If I get it right only resources from the webjars are taken into account but the spring dispatcherservlet is not available.

**Comment by janbartel** [comment]
The list of jars that is in fork.props seems ok, except you do have some in there that aren't necessary as they should be provided by the plugin, eg:
/Users/mfrey/.m2/repository/javax/annotation/javax.annotation-api/1.2/javax.annotation-api-1.2.jar
/Users/mfrey/.m2/repository/javax/el/javax.el-api/2.2.4/javax.el-api-2.2.4.jar
/Users/mfrey/.m2/repository/org/glassfish/web/javax.el/2.2.4/javax.el-2.2.4.jar

I think we'll need a small repro test project to get any further.

**Comment by Antibrumm** [comment]
I understand. This might take a while on my side as this problem is non-critical for us and is just a little gimick. 

I will update this issue with a reference to a github project once i have something. Feel free to put the issue onhold until that.

**Comment by Antibrumm** [comment]
I found the issue while creating a fork of the example project you posted.

I have split the project into a multi-module spring application and to my surprise it was all running. Both `run` and `run-forked` are fine.

Comparing the setup with my actual application i found out that the "only" difference is the declaration of the mapping of the dispatcher servlet:

`dispatcher.addMapping("/*");`

While this works in both goals it stops working for the `run-forked` goal if you change it to:

`dispatcher.addMapping("/");`

Once i found this out it's all running smoothly.

I don't know if this is a bug or just an incorrect configuration, but it seems odd that the latter is working in one of the cases.


I will keep the fork available for people trying a setup with spring and multi-modules:
https://github.com/Antibrumm/jetty-jacoco-example

**Comment by janbartel** [comment]
@Antibrumm, ah that's the lightbulb moment I needed!

Here's the situation: 

When using mvn jetty:run, or indeed running jetty from the distro, we know the origin of all of the servlets and servlet mappings because they either come from web.xml, web-fragment.xml, webdefault.xml, override-web.xml, an annotation or are programmatic.

When using mvn jetty:run-forked, it uses the quickstart mechanism that processes all of the descriptors (web.xml, web-fragment.xml, webdefault.xml and override-web.xml), and all of the annotations and then compiles them into a single unified web xml file, in this case called fork-web.xml. At that point, we've lost the origin of each of the individual elements. This is important because a programmatic ServletRegistration cannot add a mapping that conflicts with a mapping that has already been established. In this case "/" is always mapped by the default servlet in webdefault.xml. Jetty tries to take account of this and allow stuff that originated in webdefault.xml to be overridden, but we lost the info that it was relying on to identify it as coming from webdefault.xml when we generated fork-web.xml.

So that explains what is going on.

Coincidentally, I've just added more info to the generation of the quickstart web.xml (aka fork-web.xml) that may help preserve the origin and allow us to override stuff originating from webdefault.xml. I'll look into this and comment here later.
